dev-tools/create-bwc-index.py

```diff
@@ -136,11 +136,39 @@ def generate_index(client):
   num_shards = random.randint(1, 10)
   num_replicas = random.randint(0, 1)
   logging.info('Create single shard test index')
+
+  mappings = {}
+  if not version.startswith('2.'):
+    # TODO: we need better "before/onOr/after" logic in python
+
+    # backcompat test for legacy type level analyzer settings, see #8874
+    mappings['analyzer_type1'] = {
+      'analyzer': 'standard',
+      'properties': {
+        'string_with_index_analyzer': {
+          'type': 'string',
+          'index_analyzer': 'standard'
+        },
+      }
+    }
+    if not version.startswith('0.20') or version == '0.20.6':
```

Comment:
as someone who doesnt know python, i find this hard to grok. can we simplify this condition or add comments?